src/__swaps__/safe-math/SafeMath.ts

```diff
@@ -40,182 +40,316 @@ const formatResultWorklet = (result: bigint): string => {
-  const [bigInt2, decimalPlaces2] = removeDecimalWorklet(num2);
+  const [bigInt1, decimalPlaces1] = removeDecimalWorklet(num1Str);
+  const [bigInt2, decimalPlaces2] = removeDecimalWorklet(num2Str);
   const scaledBigInt1 = scaleUpWorklet(bigInt1, decimalPlaces1);
   const scaledBigInt2 = scaleUpWorklet(bigInt2, decimalPlaces2);
   const result = scaledBigInt1 + scaledBigInt2;
   return formatResultWorklet(result);
 }
 
 // Subtract function
-export function subWorklet(num1: string, num2: string): string {
+export function subWorklet(num1: string | number, num2: string | number): string {
   'worklet';
-  if (!isNumberStringWorklet(num1) || !isNumberStringWorklet(num2)) {
-    throw new Error('Arguments must be a numeric string');
+  const num1Str = toStringWorklet(num1);
+  const num2Str = toStringWorklet(num2);
+
+  if (!isNumberStringWorklet(num1Str) || !isNumberStringWorklet(num2Str)) {
+    throw new Error('Arguments must be a numeric string or number');
   }
 
-  if (isZeroWorklet(num2)) {
-    return num1;
+  if (isZeroWorklet(num2Str)) {
+    return num1Str;
   }
 
-  const [bigInt1, decimalPlaces1] = removeDecimalWorklet(num1);
-  const [bigInt2, decimalPlaces2] = removeDecimalWorklet(num2);
+  const [bigInt1, decimalPlaces1] = removeDecimalWorklet(num1Str);
+  const [bigInt2, decimalPlaces2] = removeDecimalWorklet(num2Str);
   const scaledBigInt1 = scaleUpWorklet(bigInt1, decimalPlaces1);
   const scaledBigInt2 = scaleUpWorklet(bigInt2, decimalPlaces2);
   const result = scaledBigInt1 - scaledBigInt2;
   return formatResultWorklet(result);
 }
 
 // Multiply function
-export function mulWorklet(num1: string, num2: string): string {
+export function mulWorklet(num1: string | number, num2: string | number): string {
   'worklet';
-  if (!isNumberStringWorklet(num1) || !isNumberStringWorklet(num2)) {
-    throw new Error('Arguments must be a numeric string');
+  const num1Str = toStringWorklet(num1);
+  const num2Str = toStringWorklet(num2);
+
+  if (!isNumberStringWorklet(num1Str) || !isNumberStringWorklet(num2Str)) {
+    throw new Error('Arguments must be a numeric string or number');
   }
-  if (isZeroWorklet(num1) || isZeroWorklet(num2)) {
+  if (isZeroWorklet(num1Str) || isZeroWorklet(num2Str)) {
     return '0';
   }
-  const [bigInt1, decimalPlaces1] = removeDecimalWorklet(num1);
-  const [bigInt2, decimalPlaces2] = removeDecimalWorklet(num2);
+  const [bigInt1, decimalPlaces1] = removeDecimalWorklet(num1Str);
+  const [bigInt2, decimalPlaces2] = removeDecimalWorklet(num2Str);
   const scaledBigInt1 = scaleUpWorklet(bigInt1, decimalPlaces1);
   const scaledBigInt2 = scaleUpWorklet(bigInt2, decimalPlaces2);
   const result = (scaledBigInt1 * scaledBigInt2) / BigInt(10) ** BigInt(20);
   return formatResultWorklet(result);
 }
 
 // Divide function
-export function divWorklet(num1: string, num2: string): string {
+export function divWorklet(num1: string | number, num2: string | number): string {
```

Comment:
do we want to make it "easy" to keep `number`s in the codebase? 
like we can make the user use string by not accepting number in the api or at least indicate something is up by making his code ugly having to parse before lol

Reply:
I'm just getting started to replace the unsafe Number usages inside the worklets and I'm not sure what I'm gonna find out so I'll keep it flexible like this to move faster but in the end I'd like to find all the occurrences and make it more stricter if possible